documentation/source/coroutines.rst

```diff
@@ -5,80 +5,50 @@ Coroutines
 **********
 
 Testbenches built using cocotb use coroutines. While the coroutine is executing
-the simulation is paused. The coroutine uses the :keyword:`yield` keyword to
-pass control of execution back to the simulator and simulation time can advance
-again.
+the simulation is paused. The coroutine uses the :keyword:`await` keyword to
+block on another coroutine's execution or pass control of execution back to the
+simulator, allowing simulation time to advance.
 
-Typically coroutines :keyword:`yield` a :any:`Trigger` object which
+Typically coroutines :keyword:`await` a :class:`~cocotb.triggers.Trigger` object which
 indicates to the simulator some event which will cause the coroutine to be woken
 when it occurs.  For example:
 
 .. code-block:: python3
 
-    @cocotb.coroutine
-    def wait_10ns():
+    async def wait_10ns():
         cocotb.log.info("About to wait for 10 ns")
-        yield Timer(10, units='ns')
+        await Timer(10, units='ns')
         cocotb.log.info("Simulation time has advanced by 10 ns")
 
-Coroutines may also yield other coroutines:
+Coroutines may also wait other coroutines:
```

Comment:
Either
```suggestion
Coroutines may also wait for other coroutines:
```
```suggestion
Coroutines may also wait upon other coroutines:
```
```suggestion
Coroutines may also wait on other coroutines:
```
```suggestion
Coroutines may also await other coroutines:
```

Reply:
I like the version using await, otherwise someone could find it ambiguous why we don't use the keyword when it fits perfectly.